exporter/containerimage/exptypes/types.go

```diff
@@ -24,3 +25,19 @@ type Platform struct {
 	ID       string
 	Platform ocispecs.Platform
 }
+
+type BuildInfo struct {
+	Type  BuildInfoType `json:"type,omitempty"`
+	Ref   string        `json:"ref,omitempty"`
```

Comment:
We also manage `git` and `http` sources. That's why the `Pin` attribute is used here. But if someone specifies a pseudo-reference then it will be setted in `Ref`. I have updated the example above to show you what it would look like.